doc/modules/sql/pages/querying-map.adoc

```diff
@@ -0,0 +1,197 @@
+= Querying IHMap
+
+The following sections describe how you can access Hazelcast map objects and perform queries on them.
+
+== Map Names
+
+The SQL service exposes `IHMap` objects as tables in the predefined partitioned schema using exact names. This schema is in the SQL service search path so that you can access the `IHMap` objects with or without the schema names.
+
+Schema and table names are case-sensitive; you can access the `employee` map, for example, as `employee` or `partitioned.employee`, but not as `Employee`:
+
+[source,sql]
+----
+SELECT * FROM employee
+SELECT * FROM partitioned.employee
+----
+
+== Field Names
+
+The SQL service resolves fields accessible from the SQL automatically. The service reads the first local entry pair of `IHMap` to construct the list of fields. If `IHMap` does not have local entries on the member where the query is started, then the list of fields cannot be resolved, and an exception is thrown.
+
+Field names are case-sensitive.
+
+=== Key and Value Objects
+
+An `IHMap` entry consists of a key and a value. These are accessible through the `__key` and `this` aliases. The following query returns the keys and values of all entries in a map named `employee`:
+
+[source,sql]
+----
+SELECT __key, this FROM employee
+----
+
+=== Key and Value Fields
+
+You may also access the nested fields of a key or value. The list of exposed fields depends on the serialization format, as described below:
+
+* For xref:serialization:ids.adoc[IdentifiedDataSerializable] objects, you can use public field name or getter names. See xref:hazelcast:sql:querying-maps-sql.adoc[here] for more information.
+* For xref:serialization:portable.adoc[Portable] objects, the fields written with `IPortableWriter` methods are exposed using their exact names.
+
+NOTE: You cannot query JSON fields in SQL. If you want to query JSON, see xref:using-hazelcast:queries.adoc#querying-with-json-strings[Querying with JSON Strings].
```

Comment:
This is changed since 5.1. 

"For Json objects, the fields can be accessed with JSON_QUERY(..) or JSON_VALUE(..)."
http://hazelcast.github.io/hazelcast-csharp-client/latest/doc/sql.html#querying-ihmap